### PR TITLE
Add python-keystoneclient

### PR DESCRIPTION
**Pull request by jtratner**
Stage 2 of getting recipe for stor set up :)

(I'm not clear whether this can be noarch or not)

cc @kyleabeauchamp

### CODE REVIEW

**Comment by kyleabeauchamp** [comment]
LGTM

**Comment by jtratner** [comment]
@conda-forge/staged-recipes - would you mind taking a look at this? :)

File: recipes/python-keystoneclient/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (synapticarbors) [inline recipes/python-keystoneclient/meta.yaml]: Could you switch to the new convention that uses `pip`:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L33

and then add `pip` as a `build` dependency.

File: recipes/python-keystoneclient/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+requirements:
+  build:
```
Comment (synapticarbors) [inline recipes/python-keystoneclient/meta.yaml]: Are all of these packages required to `build` the recipe, or are they just runtime requirements? If it's the latter, please remove those that aren't required by the `setup.py`.

File: recipes/python-keystoneclient/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
```
Comment (synapticarbors) [inline recipes/python-keystoneclient/meta.yaml]: Could this recipe possibly be `noarch: python`?

http://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages

File: recipes/python-keystoneclient/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    .. _Source: https://git.openstack.org/cgit/openstack/python-keystoneclient\n.. _OpenStackClient: https://pypi.python.org/pypi/python-openstackclient\n.. _How to Contribute: https://docs.openstack.org/infra/manual/developers.html\n\
+    .. _Specs: https://specs.openstack.org/openstack/keystone-specs/\n\n.. contents:: Contents:\n   :local:\n\nPython API\n----------\n\nBy way of a quick-start::\n\n    >>> from keystoneauth1.identity\
+    \ import v3\n    >>> from keystoneauth1 import session\n    >>> from keystoneclient.v3 import client\n    >>> auth = v3.Password(auth_url=\"http://example.com:5000/v3\", username=\"admin\",\n    ...\
+    \                     password=\"password\", project_name=\"admin\",\n    ...                     user_domain_id=\"default\", project_domain_id=\"default\")\n    >>> sess = session.Session(auth=auth)\n\
+    \    >>> keystone = client.Client(session=sess)\n    >>> keystone.projects.list()\n        [...]\n    >>> project = keystone.projects.create(name=\"test\", description=\"My new Project!\", domain=\"\
+    default\", enabled=True)\n    >>> project.delete()\n\n\n\n"
+  doc_url: ''
+  dev_url: ''
```
Comment (synapticarbors) [inline recipes/python-keystoneclient/meta.yaml]: Perhaps: `dev_url: https://github.com/openstack/python-keystoneclient`

File: recipes/python-keystoneclient/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    .. _Source: https://git.openstack.org/cgit/openstack/python-keystoneclient\n.. _OpenStackClient: https://pypi.python.org/pypi/python-openstackclient\n.. _How to Contribute: https://docs.openstack.org/infra/manual/developers.html\n\
+    .. _Specs: https://specs.openstack.org/openstack/keystone-specs/\n\n.. contents:: Contents:\n   :local:\n\nPython API\n----------\n\nBy way of a quick-start::\n\n    >>> from keystoneauth1.identity\
+    \ import v3\n    >>> from keystoneauth1 import session\n    >>> from keystoneclient.v3 import client\n    >>> auth = v3.Password(auth_url=\"http://example.com:5000/v3\", username=\"admin\",\n    ...\
+    \                     password=\"password\", project_name=\"admin\",\n    ...                     user_domain_id=\"default\", project_domain_id=\"default\")\n    >>> sess = session.Session(auth=auth)\n\
+    \    >>> keystone = client.Client(session=sess)\n    >>> keystone.projects.list()\n        [...]\n    >>> project = keystone.projects.create(name=\"test\", description=\"My new Project!\", domain=\"\
+    default\", enabled=True)\n    >>> project.delete()\n\n\n\n"
+  doc_url: ''
```
Comment (synapticarbors) [inline recipes/python-keystoneclient/meta.yaml]: Perhaps: `doc_url: https://docs.openstack.org/python-keystoneclient/latest/`

File: recipes/python-keystoneclient/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    - keystoneauth1 >=3.3.0
+    - debtcollector >=1.2.0
+
+test:
+  imports:
+    - keystoneclient
+
+about:
+  home: The package home page
+  license: Apache
```
Comment (synapticarbors) [inline recipes/python-keystoneclient/meta.yaml]: `license: Apache-2.0`

File: recipes/python-keystoneclient/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  license_family: APACHE
+  license_file: ''
```
Comment (synapticarbors) [inline recipes/python-keystoneclient/meta.yaml]: `license_file: LICENSE`

File: recipes/python-keystoneclient/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    \                     password=\"password\", project_name=\"admin\",\n    ...                     user_domain_id=\"default\", project_domain_id=\"default\")\n    >>> sess = session.Session(auth=auth)\n\
+    \    >>> keystone = client.Client(session=sess)\n    >>> keystone.projects.list()\n        [...]\n    >>> project = keystone.projects.create(name=\"test\", description=\"My new Project!\", domain=\"\
+    default\", enabled=True)\n    >>> project.delete()\n\n\n\n"
+  doc_url: ''
+  dev_url: ''
+
+extra:
+  recipe-maintainers:
+    - jtratner
+    - pkaleta
```
Comment (synapticarbors) [inline recipes/python-keystoneclient/meta.yaml]: @pkaleta -- are you ok being added as a co-maintainer of the `python-keystoneclient` recipe on conda-forge?

http://conda-forge.readthedocs.io/en/latest/recipe.html#maintainer-role

Comment (pkaleta) [reply recipes/python-keystoneclient/meta.yaml]: sgtm

**Pull request closed without merging**